containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,291 @@
+    ):
+        bucket_path = BucketPath.from_uri(
+            DataProductConfig(name=data_product_name).schema_path(table_name).uri
+        )
+        self.table_name = table_name
+
+        super().__init__(
+            data_product_name, logger, JsonSchemaName("schema"), bucket_path
+        )
+
+        if not self._does_data_product_metadata_exist():
+            self.logger.error("Data product metadata not yet registered.")
+            self.has_registered_data_product = False
+        else:
+            self.has_registered_data_product = True
+
+    def _does_data_product_metadata_exist(self):
+        """checks wheter data product for schema has metadata registered"""
+        data_product_metadata_path = BucketPath.from_uri(
+            DataProductConfig(name=self.data_product_name).metadata_path().uri
+        )
+        md_bucket = data_product_metadata_path.bucket
+        md_key = data_product_metadata_path.key
+        return self._check_if_metadata_or_schema_exists(
+            md_bucket, md_key, JsonSchemaName("metadata")
+        )
+
+    def convert_schema_to_glue_table_input_csv(self):
+        """
+        convert schema passed by user to glue table input so it can be used to create an athena table
+        from csv with headers.
+        """
+        if self.valid:
+            glue_schema = deepcopy(glue_csv_table_input_template)
+            parent_metadata = self._get_parent_data_product_metadata()
+            glue_schema["DatabaseName"] = self.data_product_name
+            glue_schema["TableInput"]["Name"] = self.table_name
+            glue_schema["TableInput"]["Owner"] = parent_metadata["dataProductOwner"]
+            if not parent_metadata.get("retentionPeriod", 0) == 0:
+                glue_schema["TableInput"]["Retention"] = parent_metadata.get(
+                    "retentionPeriod"
+                )
+            glue_schema["TableInput"]["Description"] = self.data_pre_convert[
+                "tableDescription"
+            ]
+
+            for col in self.data_pre_convert["columns"]:
+                glue_schema["TableInput"]["StorageDescriptor"]["Columns"].append(
+                    {
+                        "Name": col["name"],
+                        "Type": col["type"],
+                        "Comment": col["description"],
+                    }
+                )
+
+            self.data = glue_schema
+            self.logger.info("Data Product schema converted to csv glue table input")
+        else:
+            self.logger.error("schema not validated before attempted conversion")
+
+        return self
+
+    def _get_parent_data_product_metadata(self) -> Dict:
+        data_product_metadata_path = BucketPath.from_uri(
+            DataProductConfig(name=self.data_product_name)
+            .metadata_path(version=self.data_product_version)
+            .uri
+        )
+
+        metadata = read_json_from_s3(data_product_metadata_path.uri)
```

Comment:
Suggestion: We could perhaps implement this as a static `load()` method on the base class if we want it to support reads in addition to writes.

Reply:
I think reads will be useful, especially when the class is extended to deal with updates

Reply:
Yeah good point - we will need it to be able to classify major/minor